pkg/controller/datadogagentdeployment/datadogagentdeployment_controller.go

```diff
@@ -263,3 +279,86 @@ func (r *ReconcileDatadogAgentDeployment) updateStatusIfNeeded(logger logr.Logge
 
 	return result, currentError
 }
+
+// sendDatadogMetrics establishes a connection with the Datadog API
+// it starts sending deployment metrics once the connection is validated
+func (r *ReconcileDatadogAgentDeployment) sendDatadogMetrics(ctx context.Context) {
+	retryTicker := time.NewTicker(r.datadogMetricsForwarder.RetryInterval)
+RetryLoop:
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-retryTicker.C:
+			dad, err := r.listDatadogAgentDeployment()
```

Comment:
If you pass the `ctx` down to listDatadogAgentDeployment it might help with getting rid of `context.TODO()` in that func.